token_auth/serializers/register.py

```diff
@@ -0,0 +1,45 @@
+from django.contrib.auth import get_user_model
+from rest_framework import serializers
+from rest_framework_jwt.serializers import PasswordField
+
+User = get_user_model()
+
+class RegisterSerializer(serializers.ModelSerializer):
+
+    password = PasswordField(
+        write_only=True,
+        required=True
+    )
+    password2 = PasswordField(
+        write_only=True,
+    )
+
+    class Meta:
+        model = User
+        fields = [
+            "username",
+            "email",
+            "password",
+            "password2",
+        ]
+
+    def create(self, validated_data):
+
+        username = validated_data["username"]
+        email = validated_data["email"]
+        password = validated_data["password"]
+        password2 = validated_data["password2"]
+
+        if (email and User.objects.filter(email=email).exclude(username=username).exists()):
+            raise serializers.ValidationError(
+                {"email": "Email addresses must be unique."})
+
+        if password != password2:
+            raise serializers.ValidationError(
+                {"password": "The two passwords differ."})
+
+        user = User(username=username, email=email)
+        user.set_password(validated_data['password'])
```

Comment:
you already have this in a var

Reply:
@Monal5031 not clear?

Reply:
`password` variable is already initialized at start of this function